app/jobs/metais/diff_consecutive_versions_job.rb

```diff
@@ -0,0 +1,22 @@
+class Metais::DiffConsecutiveVersionsJob < ApplicationJob
+  queue_as :metais
+
+  OBSERVED_ATTRIBUTES = ['nazov', 'kod_metais', 'typ_investicie', 'prijimatel', 'faza_projektu', 'program', 'popis', 'datum_zacatia', 'termin_ukoncenia', 'schvalovaci_proces', 'zdroj', 'financna_skupina', 'suma_vydavkov', 'rocne_naklady', 'ref_id', 'status', 'zmena_stavu', 'schvalene_rocne_naklady', 'schvaleny_rozpocet', 'datum_nfp', 'link_nfp', 'vyhlasenie_vo', 'vo', 'zmluva_o_dielo', 'zmluva_o_dielo_crz']
+
+  def perform(version)
+    ActiveRecord::Base.transaction do
+      result = Metais::ProjectVersion.previous(version)
+      return if result.blank?
+      previous = result.first
+
+      OBSERVED_ATTRIBUTES.each do |key|
+        Metais::ProjectChange.create(
+          project_version: version,
+          atribut: key,
+          predchadzajuca_hodnota: previous[key].to_s,
+          nova_hodnota: version[key].to_s
```

Comment:
tuto si drzme anglicke nazvy. Je to nasa interna tabulka, ma zmysel sa drzat rails konvencii. Inak zvazovali sme, ze dame do jsonb vsetky zmenene atributy a nie takuto EAV (https://en.wikipedia.org/wiki/Entity%E2%80%93attribute%E2%80%93value_model)  tabulku?

Reply:
@jsuchal keď si napísal o tom jsonb, pozrel som na to. Dá sa to spraviť, ale v metabase sa podľa mňa bude jednoduchšie pracovať s EAV-like tabuľkou. S jsonb tam sú tam vraj ešte stále nejaké problémy.

Nevidím ale výhody jsonb oproti tomuto EAV-like. Náš usecase je, že väčšinou chceme zmeny filtrovať podľa zmeneného atribútu. Pri jsonb by trebalo teda vždy pozerať, či obsahuje takýto key. A na druhej strane, my nepotrebujeme pristupovať k tým atribútom cez túto eav tabuľku, takže to veľmi nekazí model.

Čo sa týka anglických názvov, dal som, ale `attribute` som musel dať na `field`, lebo `attribute` je známe slovo pre ruby.

Reply:
@celuchmarek skusme teda takto.